app.py

```diff
@@ -36,13 +36,113 @@ def user_index():
     userFirstRow = query.first()
     userData = pd.DataFrame(data)
 
-    #TODO Resolve if we need all trainings in index.html
-    trainings = db.session.query(Event.log_time, Event.event_type, Card.user_id, User.first_name).select_from(Event).join(Card, full=True).join(User, full=True).filter(User.id==userId,Event.event_type=="start").order_by(Event.log_time.desc()).all()
+
+    trainings = db.session.query(Event.log_time, Event.event_type, Card.user_id, User.first_name).select_from(Event).join(Card, full=True).join(User, full=True).filter(User.id==userId,Event.event_type=="start").order_by(Event.log_time.desc()).limit(USER_TRAININGS_LIMIT)
     userTrainings = pd.DataFrame(trainings)
 
     isTraining = userData['event_type'][0]=='start'
 
-    return render_template('main/index.html', userData=userData, userFirstRow=userFirstRow, isTraining=isTraining, userTrainings=userTrainings, userTrainingsLimit=USER_TRAININGS_LIMIT)
+    return render_template('main/index.html', userData=userData, userFirstRow=userFirstRow, isTraining=isTraining, userTrainings=userTrainings)
+
+
+@app.route('/users/list', methods=['GET'])
+def user_list():
+    users = db.session.query(
+        Card.user_id,
+        User.first_name,
+        User.last_name,
+        Card.id
+    ).select_from(User).join(Card, full=True).all()
+    usersData = pd.DataFrame(users)
+    return render_template('main/users.html', usersData = usersData)
+
+
+@app.route('/users/add', methods=['GET', 'POST'])
+def user_add():
+    if request.method == 'GET':
+        return render_template('main/create_user.html')
+    
+    if request.method == 'POST':
+        user_first_name = request.form['user_first_name']
+        user_last_name = request.form['user_last_name']
+
+        new_user = User(first_name = user_first_name, last_name = user_last_name)
+        db.session.add(new_user)
+        db.session.commit()
+        return redirect(url_for('user_list'))
+
+@app.route('/users/assign', methods=['GET', 'POST'])
+def user_assign():
+    if request.method == 'GET':
+        users = db.session.query(
+            User.id,
+            User.first_name,
+            User.last_name
+        ).select_from(User).all()
+
+        cards = db.session.query(
+            Card.id,
+            Card.user_id
+        ).select_from(Card).all()
+
+        usersData = pd.DataFrame(users)
+        cardsData = pd.DataFrame(cards)
+
+        return render_template('main/assign_user.html', usersData=usersData, cardsData=cardsData)
+    
+    if request.method == 'POST':
+        
```

Comment:
Redundant new lines.

Reply:
Didn't fix this one. Please fix it while working on backend update.